opentasks/src/main/java/org/dmfs/tasks/share/TaskBindings.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2017 dmfs GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.dmfs.tasks.share;
+
+import android.support.annotation.Nullable;
+import android.text.format.Time;
+
+import org.dmfs.iterators.Function;
+import org.dmfs.tasks.R;
+import org.dmfs.tasks.model.ContentSet;
+import org.dmfs.tasks.model.Model;
+import org.dmfs.tasks.model.TaskFieldAdapters;
+
+import au.com.codeka.carrot.Bindings;
+import au.com.codeka.carrot.bindings.SingletonBindings;
+
+
+/**
+ * {@link Bindings} for accessing values from a task.
+ * See the <code>switch-case</code> for the supported properties.
+ * <p>
+ * Uses a {@link Function} of {@link Time} to {@link String} received in the constructor to format the times.
+ *
+ * @author Gabor Keszthelyi
+ */
+public final class TaskBindings implements Bindings
+{
+    private final Bindings mDelegate;
+
+
+    public TaskBindings(ContentSet contentSet, Model model)
+    {
+        mDelegate = new SingletonBindings("$task", new SuffixBindings(contentSet, model));
+    }
+
+
+    @Nullable
+    @Override
+    public Object resolve(String key)
+    {
+        return mDelegate.resolve(key);
+    }
+
+
+    @Override
+    public boolean isEmpty()
+    {
+        return mDelegate.isEmpty();
+    }
+
+
+    private static final class SuffixBindings implements Bindings
```

Comment:
This should be the `TaskBindings` class. The `SingletonBindings` of your `TaskBindings` should be created in `ShareTasktext.OutputFactory.create()` (just like you bind the tformat function).

Reply:
Right, I guess it's better if prefix is defined independently at the place of usage. Updated.